app/routes/u.github/$slug.tsx

```diff
@@ -0,0 +1,121 @@
+import { json, LoaderFunction, redirect } from "@remix-run/node";
+import { Link, Outlet, useCatch, useLoaderData } from "@remix-run/react";
+import { FC } from "react";
+import RootLayout from "~/components/RootLayout";
+import { getUserProfileBySlug } from "~/database.server";
+import { UserProfile } from "~/types";
+
+export const loader: LoaderFunction = async ({ params }) => {
+  const slug = params.slug;
+
+  if (typeof slug !== "string") {
+    return redirect("/");
+  }
+
+  const userProfile = await getUserProfileBySlug(slug);
+
+  if (!userProfile) {
+    throw new Response("Not Found", {
+      status: 404,
+    });
+  }
+
+  return json(userProfile);
```

Comment:
If we find a matching user profile in the database, we send it to the component to get rendered. Otherwise, we throw a 404...